script_flow.go

```diff
@@ -17,46 +17,43 @@
 package edgedb
 
 import (
-	"context"
-	"net"
-
-	"github.com/edgedb/edgedb-go/protocol/buff"
-	"github.com/edgedb/edgedb-go/protocol/message"
+	"github.com/edgedb/edgedb-go/internal/buff"
+	"github.com/edgedb/edgedb-go/internal/message"
 )
 
-func (c *baseConn) scriptFlow(
-	ctx context.Context,
-	conn net.Conn,
-	query string,
-) error {
-	buf := buff.New(nil)
-	buf.BeginMessage(message.ExecuteScript)
-	buf.PushUint16(0) // no headers
-	buf.PushString(query)
-	buf.EndMessage()
+func (c *baseConn) scriptFlow(r *buff.Reader, query string) error {
+	c.writer.BeginMessage(message.ExecuteScript)
+	c.writer.PushUint16(0) // no headers
+	c.writer.PushString(query)
+	c.writer.EndMessage()
 
-	err := c.writeAndRead(ctx, buf.Unwrap())
-	if err != nil {
-		return err
+	if e := c.writer.Send(c.conn); e != nil {
+		return e
 	}
 
-	for buf.Next() {
-		switch buf.MsgType {
+	var err error
+	done := buff.NewSignal()
+
+	for r.Next(done.Chan) {
+		switch r.MsgType {
 		case message.CommandComplete:
-			buf.PopUint16() // header count (assume 0)
-			buf.PopBytes()  // command status
+			r.Discard(2) // header count (assume 0)
```

Comment:
Please add an internal function similar to `reject_headers` in edgedb-python

Reply:
:+1: created issue here https://github.com/edgedb/edgedb-go/issues/40